contracts/cw-named-groups/src/msg.rs

```diff
@@ -0,0 +1,58 @@
+use std::collections::HashSet;
+
+use cosmwasm_std::Addr;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Group {
+    pub name: String,
+    pub addresses: HashSet<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InstantiateMsg {
+    pub groups: Option<Vec<Group>>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    Add {
+        group: String,
+        addresses: Option<Vec<String>>,
+    },
+    Remove {
+        group: String,
+        addresses: Option<Vec<String>>,
+    },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum QueryMsg {
+    Dump {},
+    ListGroups { address: String },
+    ListAddresses { group: String },
```

Comment:
We should paginate these queries. Dump is OK to run out of gas, but for these it would be nice if they didn't. See `ListProposals` in `contracts/cw-governance` for an example of doing this.